src/Opdex.Platform.Infrastructure/Clients/CirrusFullNodeApi/Handlers/Tokens/CallCirrusGetInterfluxTokenContractSummaryQueryHandler.cs

```diff
@@ -0,0 +1,53 @@
+using MediatR;
+using Opdex.Platform.Common.Constants.SmartContracts.Tokens;
+using Opdex.Platform.Common.Models;
+using Opdex.Platform.Domain.Models.Tokens;
+using Opdex.Platform.Infrastructure.Abstractions.Clients.CirrusFullNodeApi.Models;
+using Opdex.Platform.Infrastructure.Abstractions.Clients.CirrusFullNodeApi.Modules;
+using Opdex.Platform.Infrastructure.Abstractions.Clients.CirrusFullNodeApi.Queries.Tokens;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Opdex.Platform.Infrastructure.Clients.CirrusFullNodeApi.Handlers.Tokens;
+
+public class CallCirrusGetInterfluxTokenContractSummaryQueryHandler
+    : IRequestHandler<CallCirrusGetInterfluxTokenContractSummaryQuery, InterfluxTokenContractSummary>
+{
+    private readonly ISmartContractsModule _smartContractsModule;
+
+    public CallCirrusGetInterfluxTokenContractSummaryQueryHandler(ISmartContractsModule smartContractsModule)
+    {
+        _smartContractsModule = smartContractsModule ?? throw new ArgumentNullException(nameof(smartContractsModule));
+    }
+
+    public async Task<InterfluxTokenContractSummary> Handle(CallCirrusGetInterfluxTokenContractSummaryQuery request, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var ownerRequest = new LocalCallRequestDto(request.Token, request.Token,
+                $"get_{InterfluxTokenConstants.Properties.Owner}", request.BlockHeight);
+            var ownerResponse = await _smartContractsModule.LocalCallAsync(ownerRequest, cancellationToken);
+            var owner = ownerResponse.DeserializeValue<Address>();
```

Comment:
Since this is making multiple calls and we're checking this for all created tokens, we should fail fast rather than throw exceptions within `summary.SetInterfluxDetails` to catch then return null.

If the Owner is null or Address.Zero - return null immediately
if the NativeChain isn't valid - return null immediately

Reply:
Resolved this and while at it, added a `TryDeserializeValue` instead so we're not throwing exceptions. Although, I'm not returning null if the Owner is `Address.Zero` since there's a method on the contract to revoke ownership, which sets it to `Address.Zero`. In which case we'd still want to index this